log4j-core/src/main/java/org/apache/logging/log4j/core/impl/LogEventFactory.java

```diff
@@ -28,8 +28,21 @@
 /**
  *
  */
-public interface LogEventFactory {
+public interface LogEventFactory extends LocationAwareLogEventFactory {
```

Comment:
@rgoers I'd appreciate your input on this part of the change. Now that we're using java 8 we can simplify call-sites to invoke the location-aware implementation. It's possible though unlikely that implementations could exist specifically to avoid capturing location information, but our logger configs support configuring that directly.